docs/developer-guide/working-with-plugins.md

```diff
@@ -18,6 +18,21 @@ module.exports = {
 };
 ```
 
+### Preprocessors in Plugins
+
+You can also create plugins that would tell ESLint how to process files other then JavaScript. In order to create a
+preprocessor, object that is exported from your module has to conform to the following interface:
+
+```js
+{
+    preprocess: function(string) { return [string] } //Takes text of the file and returns array of texts to be processed
+    postprocess: function([[Message]]) { return [Message] //Takes array of array of error messages, one for each text
```

Comment:
`result` object is not yet created at this stage. This is because preprocesss can return multiple blocks of code, and postprocess has to aggregate them into a single set of messages. Once postprocess is complete, cli will create `result` object out of return value (this way it will align with what happens to standard javascript files).